### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by TaslimOwolarafe**
I haven't tested this at all.

I have this issue:

`
ERROR:django.request:Internal Server Error: /talents/
Traceback (most recent call last):
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)     
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^     
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\views\decorators\csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\rest_framework\viewsets.py", line 103, in view
    self = cls(**initkwargs)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django_elasticsearch_dsl_drf\viewsets.py", line 160, in __init__
    self.client = connections.get_connection(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch_dsl\connections.py", line 109, in get_connection
    return self.create_connection(alias, **self._kwargs[alias])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch_dsl\connections.py", line 84, in create_connection
    conn = self._conns[alias] = self.elasticsearch_class(**kwargs)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch\_sync\client\__init__.py", line 194, in __init__
    raise ValueError("Either 'hosts' or 'cloud_id' must be specified")
#### ValueError: Either 'hosts' or 'cloud_id' must be specified
`

### CODE REVIEW

**Comment by TaslimOwolarafe** [comment]
Sorry I combined another feature on this branch.
That was a mistake.

File: server/match/urls.py [hunk 1]
```diff
@@ -20,5 +20,10 @@
 
 urlpatterns = [
     path("admin/", admin.site.urls),
+    path('users/', include('users.urls')),
     path(r"", include(talent_router.urls))
 ]
+
+urlpatterns+=[
+    path('api-auth/', include('rest_framework.urls'),),
```
Comment (Taycode) [inline server/match/urls.py]: do we need this ??

Comment (TaslimOwolarafe) [reply server/match/urls.py]: Not anymore, I guess.
I think I used it to test the fields required "username" or "email".

but i had to use `{'username': email@mail.blaah, 'password': 'abc'}` for **"/users/auth/"** (with username: email).

Comment (Taycode) [reply server/match/urls.py]: so we are not using email password ??

Comment (TaslimOwolarafe) [reply server/match/urls.py]: We are. I tried with 'email' but that didn't work until I put username.

But it takes in the email. The field name is just username.

I'll find a fix for that.

File: server/users/views.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django.shortcuts import render
+from rest_framework.authtoken.views import ObtainAuthToken
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+
+class CustomAuthToken(ObtainAuthToken):
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data,
```
Comment (Taycode) [inline server/users/views.py:11]: I think you can just create your serializer and use email as the field instead of username

Comment (Taycode) [reply server/users/views.py:11]: [This](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/views.py) is the serializer they used, we can write our own @TaslimOwolarafe

File: server/match/urls.py (hunk 1 above)
Comment (Taycode) [reply server/match/urls.py]: found a fix for you, its in the comment [below](https://github.com/Taycode/match-monorepo/pull/6#discussion_r1569141136).

**Comment by TaslimOwolarafe** [comment]
```
>>> import requests
>>> endpoint="http://localhost:7222/users/auth/"
>>> ar=requests.post(endpoint, json={'email':'tas@mail.com','password':'taslim'}) 
>>> ar.json()
{'token': '83a8904f0fbc385ac1e681cd7fb3c6b9b90ba6e3', 'user_id': 1}
```

@Taycode

**Comment by TaslimOwolarafe** [comment]
> I haven't tested this at all.
> 
> I have this issue:
> 
> ` ERROR:django.request:Internal Server Error: /talents/ Traceback (most recent call last): File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner response = get_response(request) ^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response response = wrapped_callback(request, *callback_args, **callback_kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django\views\decorators\csrf.py", line 56, in wrapper_view return view_func(*args, **kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\rest_framework\viewsets.py", line 103, in view self = cls(**initkwargs) ^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\django_elasticsearch_dsl_drf\viewsets.py", line 160, in **init** self.client = connections.get_connection( ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch_dsl\connections.py", line 109, in get_connection return self.create_connection(alias, **self._kwargs[alias]) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch_dsl\connections.py", line 84, in create_connection conn = self._conns[alias] = self.elasticsearch_class(**kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Users\USER\Desktop\match-monorepo\env\Lib\site-packages\elasticsearch_sync\client__init__.py", line 194, in **init** raise ValueError("Either 'hosts' or 'cloud_id' must be specified")
> 
> #### ValueError: Either 'hosts' or 'cloud_id' must be specified
> `


And some of those problems i was having was due to the changing container name in docker, so  I had to do this:

services:
  db:
    `container_name: db`
    image: postgres:latest
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: match
      POSTGRES_USER: match
      POSTGRES_PASSWORD: match
    ports:
      - "5433:5432"

  elasticsearch:
    `container_name: elasticsearch`
    image: docker.elastic.co/elasticsearch/elasticsearch:8.10.0
    environment:
    `...`

**Review by Taycode — APPROVED** [review]

**Pull request merged**